src/main/java/com/apeer/sdk/impl/ApeerDevKit.java

```diff
@@ -0,0 +1,144 @@
+     * @throws ApeerInputException When key could not be found or type is not supported
+     */
+    @SuppressWarnings("unchecked")
+    public <T> T getInput(String key, Class<T> type) throws ApeerInputException {
+        if (!inputJson.has(key)) {
+            throw new ApeerInputException("Could not find key \"" + key + "\" in inputs");
+        }
+
+        if (type == String.class) {
+            return (T) inputJson.getString(key);
+        } else if (type == Integer.class || type == int.class) {
+            return (T) (Integer) inputJson.getInt(key);
+        } else if (type == Double.class || type == double.class) {
+            return (T) (Double) inputJson.getDouble(key);
+        } else if (type == Boolean.class || type == boolean.class) {
+            return (T) (Boolean) inputJson.getBoolean(key);
+        } else if (type.isArray()) {
+            var jsonItems = inputJson.getJSONArray(key);
+            T items = (T) Array.newInstance(type.getComponentType(), jsonItems.length());
+            for (int i = 0; i < jsonItems.length(); i++) {
+                Array.set(items, i, jsonItems.get(i));
+            }
+            return items;
+        } else {
+            throw new ApeerInputException("Input type \"" + type.toString() + "\" is not allowed. Use String, Integer, Double or Boolean");
+        }
+    }
+
+    /**
+     * Sets the output that will be written to output_params_file
+     *
+     * @param key   The output key as defined in the module_specification.json of your module
+     * @param value The value that will be written to the associated key. Can be any type that is supported by JSON but
+     *              must correspond to the type as specified in the module_specification.json of your module
+     * @throws ApeerOutputException When the value could not be parsed to JSON
+     */
+    public void setOutput(String key, Object value) throws ApeerOutputException {
+        try {
+            outputJson.put(key, value);
+        } catch (JSONException ex) {
+            throw new ApeerOutputException("Could not set output \"" + key + "\"", ex);
+        }
+    }
+
+    /**
+     * Sets a file output that will be written to output_params_file. Also copies the file to the output folder of your
+     * module as required by the APEER environment
+     *
+     * @param key            The output key as defined in the module_specification.json of your module
+     * @param outputFilePath The relative path to your file as you saved it
+     * @throws ApeerOutputException When the value could not be parsed to JSON
+     */
+    public void setFileOutput(String key, String outputFilePath) throws ApeerOutputException {
+        if (!outputFilePath.startsWith("/output/")) {
+            var targetPath = "/output/" + outputFilePath;
+            fileOutputWriter.moveFile(Path.of(outputFilePath), Path.of(targetPath));
+            outputFilePath = targetPath;
```

Comment:
This method doesn't deal with array of files

Reply:
dang!

Reply:
fixed!